### PR TITLE
create comprehensive time off requests dump

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Creates comprehensive time off requests task that dumps data to exported CSV
#### Where should the reviewer start?
The flow of the file is written out at the top of the module export, functions are written below in order of execution. 
#### How should this be manually tested?
execute module.exports.timeOffRequests() with or without designating params 
#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-181
#### Questions:
Why don't we use config.WhenIWork in tasks?

### CODE REVIEW

File: tasks/timeOffRequests.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  getRequestsPromise(reqParams)
+  .then(mergeResponses)
+  .then(reformatForCSV)
+  .then(exportCSV)
+  .catch(function(err){
+    CONSOLE_WITH_TIME(err);
+  });
+
+  function createDatesParams(endTimeDate, totalDays) {
+    var end = endTimeDate ? moment(endTimeDate, date_format).format(date_format) : moment().format(date_format);
```
Comment (jrauschenberg) [inline tasks/timeOffRequests.js]: Wouldn't it be cleaner to set "end" above where you set totalDays? Then you can confidently pass both into createDatesParams.

File: tasks/timeOffRequests.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+var WhenIWork = require('wheniwork-unofficial');
+var api = new WhenIWork(KEYS.wheniwork.api_key, KEYS.wheniwork.username, KEYS.wheniwork.password);
```
Comment (jrauschenberg) [inline tasks/timeOffRequests.js]: Should we use the config WiW here so you can test this if needed?

**Comment by jrauschenberg** [comment]
I tested manually on my local computer and it worked like a charm. Other than those couple comments, I think it's good to go! 👍

File: tasks/timeOffRequests.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+var moment = require('moment');
+var date_format = 'YYYY-MM-DD HH:mm:ss';
+
+var fs = require('fs');
+
+/* 
+  timeOffRequests: Returns a csv (written to file) of timeOffRequests for designated timeframe.
```
Comment (tongxiang) [inline tasks/timeOffRequests.js]: 💯  for good documentation!

**Comment by tongxiang** [comment]
👍 good to merge @apackin.

**Comment by tongxiang** [comment]
(How to run with console function: `node console timeOffRequests timeOffRequests`)

**Comment by apackin** [comment]
@tongxiang should I change the naming so that timeOffRequests doesn't have to be typed twice somehow?

**Pull request merged**